ios/RNSpotifyRemoteAppRemote.m

```diff
@@ -453,32 +453,26 @@ -(void)resetEventSubscriptions{
     }];
 }
 
-#pragma mark - RNEventConformer Implementation
+// Will be called when this module's first listener is added.
+-(void)startObserving {
+    // Set up any upstream listeners or background tasks as necessary
+    [_eventSubscriptions setValue:@YES forKey:eventType];
```

Comment:
What is `eventType` here?  Does the RCTEventEmitter give an event type when someone starts observing?

Reply:
The docs also talk briefly about implementing [`supportedEvents`](https://reactnative.dev/docs/native-modules-ios#sending-events-to-javascript), do we need to implement that here?

Reply:
> What is `eventType` here? Does the RCTEventEmitter give an event type when someone starts observing?
> 
> Ideally, we'd like to lazily subscribe to the underlying spotify events to avoid doing too much work. It seems in the case of the RCTEventEmitter that it only looks at whether or not there is a subscriber in general, not necessarily **what** they are subscribed to.

You are right. I did not really understand the logic there so I blindly copy those over. I guess in this case, we must setValue for all events but that effectively remove the purpose of the lazy subscription.  I will try to look into the doc to see if we can see the event name being subscribed to.

We do need the `supportedEvents`, so I will fix it in the meantimes.

Reply:
We might be able to do something based on the RCTEventEmitter class.

https://github.com/facebook/react-native/blob/8bd3edec88148d0ab1f225d2119435681fbbba33/React/Modules/RCTEventEmitter.m#L100-L133

 We can try to "call the parent method" (not sure it  applies in obj C). I don't really know objective C though, so it might be hard for me.